### PR TITLE
compat.pl -> cppo

### PR DESCRIPTION
**Pull request by liyishuai**
depends on ocaml-community/cppo#76

### CODE REVIEW

**Comment by Lysxia** [comment]
PERFECT.

File: coq-simple-io.opam [hunk 1]
```diff
@@ -21,6 +21,7 @@ depends: [
   "coq" {>= "8.11~"}
   "coq-ext-lib" {>= "0.10.0"}
   "ocamlbuild" {with-test & >= "0.9.0"}
+  "cppo" {build}
```
Comment (Lysxia) [inline coq-simple-io.opam:24]: This will need a lower bound.

Comment (liyishuai) [reply coq-simple-io.opam:24]: Gonna wait for cppo release it.

**Review by Lysxia — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -20,12 +20,12 @@ uninstall: $(MAKEFILE_COQ)
...
 
-COMPATFILES:=plugin/compat.ml
+COMPATFILES:=plugin/coqsimpleio.mlg
 
 compat: $(COMPATFILES)
 
-$(COMPATFILES): compat.pl
-	$(V)perl -- compat.pl $(COMPATFILES)
+%: %.cppo
+	$(V)cppo -V COQ:$(word 1, $(shell coqc -print-version)) -n -o $@ $^
```
Comment (Lysxia) [inline Makefile:28]: Is there a reason for setting `-n`? It makes sense for Coq files, but `-s` seems nicer for OCaml files.

Comment (liyishuai) [reply Makefile:28]: Does cppo print the right file without `-n`? If yes I don't mind unsetting this flag.

Comment (Lysxia) [reply Makefile:28]: Ah, nevermind, it goes through coqpp which doesn't understand line directives...

**Pull request merged**